utils/python/CIME/case_setup.py

```diff
@@ -84,7 +84,7 @@ def case_setup(caseroot, clean=False, test_mode=False, reset=False):
 
     cimeroot = os.environ["CIMEROOT"]
 
-    case = Case()
+    case = Case(caseroot)
```

Comment:
Take out the caseroot argument and put this above the construction:

    import pdb
    pdb.set_trace()

And then step into the constructor and see what's going on. Pdb is very powerful and you can print any valid python expression as if it occurred in the current stack frame.